pkg/trace/api/tracer_flare_test.go

```diff
@@ -0,0 +1,105 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package api
+
+import (
+	"io"
+	"mime/multipart"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func getTraceFlareBody(multipartBoundary string) io.ReadCloser {
+	bodyReader, bodyWriter := io.Pipe()
+
+	writer := multipart.NewWriter(bodyWriter)
+	writer.SetBoundary(multipartBoundary)
+
+	go func() {
+		defer bodyWriter.Close()
+		defer writer.Close()
+
+		archive, err := os.Create("flare.zip")
```

Comment:
nit: we should probably use `CreateTemp` here since it's a temporary file

Reply:
`CreateTemp` adds a random string to the file name, I think it will make it harder to assert the file. Is there any other reason why we prefer using `CreateTemp` here?

Reply:
Discussed offline we decided to use `Create` instead of `CreateTemp` here. 
The reason behind this choice is that `CreateTemp` adds a random string to the file name, which is more beneficial when testing multiple go routines to avoid them selecting the same file. However it is not needed here.